[BACK-3493] Add sample interval filter

darinkrauss opened the pull request:
- Add sample interval filter
- https://tidepool.atlassian.net/browse/BACK-3493

@jh-bate Not tested, but I believe this is correct. When the `sampleFilterMinimum` query parameter is specified and greater than 0, then will only return `cbg` data that is greater than or equal to `sampleFilterMinimum`.

jh-bate reviewed (approved):
added some tests and made small fix

jh-bate commented:
/deploy int

jh-bate commented:
/deploy int

darinkrauss commented:
@jh-bate Thanks for making the fix and adding tests. I added some minor fixes and updates with the latest commit. Mainly I fixed the number of milliseconds in 1 minute and added `cbg` data without `sampleInterval`.

Would you mind reviewing and approving, assuming you don't find any issues.
